todoapp/src/components/Todo.tsx

```diff
@@ -1,182 +1,87 @@
-      position: 'relative',
-      "@media (min-width: 768px)": {
-        width: "40%",
-      },
-      "@media (min-width: 1592px)": {
-        width: "30%",
-      },
-    }
+type Todo = {
+  id: string;
+  text: string;
+  completed: boolean
+}
 
-  },
-  firststack: {
-    "&&": {
-      marginBottom: '40px'
-    }
-  },
-  headings: {
-    "&&": {
-      marginLeft: '50px',
-      marginTop: '15px',
-      fontWeight: 'bold',
-    }
-  },
-  subheadings: {
-    "&&": {
-      marginLeft: '60px',
-      mb: '15px',
-      fontWeight: 'bold',
-    }
-  },
-  hr: {
-    "&&": {
-      margin: 'center',
-      marginInlineStart: '7%',
-      marginInlineEnd: '7%',
-    }
-  },
-  card: {
-    "&&": {
-      background: '#ff5252',
-      maxHeight: '300px',
-      overflowY: 'auto',
-    }
-  },
-  cardContent: {
-    "&&": {
-      display: 'flex',
-      alignItems: 'center',
-      justifyContent: 'space-between',
-      background: 'rgba(255, 205, 210, 0.25)',
-    }
-  },
-  stacktwo: {
-    "&&": {
-      position: "absolute",
-      width: "100%",
-      left: 0,
-    }
-  },
-  taskName: {
-    "&&": {
-      color: "white",
-      marginLeft: "50px",
-    }
-  },
-  newTodo: {
-    "&&": {
-      position: 'absolute',
-      bottom: '50px',
-      width: '100%',
-      flexDirection: 'column',
-      marginLeft: '60px',
-      marginRight: '50px',
-      overflowY: 'auto',
-    }
-  },
-  todoHeading: {
-    "&&": {
-      mb: '5px',
-      fontWeight: 'bold',
-    }
-  },
-  todoContainer: {
-    "&&": {
-      gap: '0.4rem',
-      marginRight: '60px',
-      display: 'flex',
-    }
-  },
-  todoTextField: {
-    "&&": {
-      width: '60%',
-      fontWeight: 'bold',
-      borderRadius: 0,
-      '& .MuiOutlinedInput-root': {
-        backgroundColor: 'white',
-        '&:hover fieldset': {
-          borderColor: 'white',
-        },
-        '&.Mui-focused fieldset': {
-          borderColor: '#ff5252',
-        },
-      },
-      "@media (max-width: 600px)": {
-        todoTextField: {
-          width: '50%',
-        }
-      }
-    }
-  },
-  todoButton: {
-    "&&": {
-      backgroundColor: '#ff5252',
-      borderColor: 'white',
-      color: 'white',
-      fontWeight: 'bold',
-      borderRadius: 0,
-      '&:hover': {
-        backgroundColor: 'black',
-      },
-      '&:focus': {
-        outlineColor: '#ff5252',
-      },
-      "@media (max-width: 600px)": {
-        todoButton: {
-          width: '100%',
+const Todos = () => {
+  const isSmallScreen = useMediaQuery('(max-width:600px)');
+  const classes = useStyles();
+  const { tasksList } = useGlobalState();
+  const text = useHookstate("");
+  const selectedTodoState = useHookstate<string | null>(null);
+
+  const handleChange = (value: string) => {
+    text.set(value);
+  }
+  const createTodo = () => {
+    const prevTodos = tasksList.get();
+    if (selectedTodoState.value && prevTodos) {
+      const updatedTodos = prevTodos.map((todo) => {
```

Comment:
this logic :  text.get().trim()  is being repeat in this function so add this in the start of the function like this
 const newText = text.get().trim();
 and then use it where needed